src/aspire/utils/blk_diag_func.py

```diff
@@ -1,15 +1,29 @@
+"""
```

Comment:
All of these should probably be merged into a class with operator overloading. Let's make an issue.

Reply:
Made the issue #111.